web/tests/filter.test.js

```diff
@@ -785,6 +811,18 @@ test("canonicalization", () => {
     term = Filter.canonicalize_term({operator: "has", operand: "reactions"});
     assert.equal(term.operator, "has");
     assert.equal(term.operand, "reaction");
+
+    const channel_term = {operator: "channel", operand: ""};
+    const topic_term = {operator: "topic", operand: ""};
+
+    const term_1 = Filter.canonicalize_narrow([{operator: "with", operand: 12}]);
+    const term_2 = Filter.canonicalize_narrow([topic_term, {operator: "with", operand: 12}]);
+    const term_3 = Filter.canonicalize_narrow([channel_term, {operator: "with", operand: 12}]);
+    const terms = [term_1, term_2, term_3];
+
+    for (const term of terms) {
+        assert.deepEqual(term, [channel_term, topic_term, {operator: "with", operand: 12}]);
+    }
```

Comment:
This block should probably be moved to be its own test, since it's basically unrelated to `canonicalize_term`.

Reply:
``` diff
diff --git a/web/tests/filter.test.js b/web/tests/filter.test.js
index 52f61aa4dc..bb2e371784 100644
--- a/web/tests/filter.test.js
+++ b/web/tests/filter.test.js
@@ -811,13 +811,15 @@ test("canonicalization", () => {
     term = Filter.canonicalize_term({operator: "has", operand: "reactions"});
     assert.equal(term.operator, "has");
     assert.equal(term.operand, "reaction");
+});
 
+test("ensure_channel_topic_terms", ({override}) => {
     const channel_term = {operator: "channel", operand: ""};
     const topic_term = {operator: "topic", operand: ""};
 
```